### DISCUSSION TITLE
Taking a task to complete

### DISCUSSION THREAD

**Discussion by apinanyogaratnam** [BushelScript/BushelScript#9] [General]
I think I can be able to implement the word completion suggestions which seems easy enough. 
<img width="401" alt="Screen Shot 2021-06-28 at 6 18 56 PM" src="https://user-images.githubusercontent.com/72412733/123710776-4e1c1500-d83d-11eb-8236-fefd10745bab.png">
I plan to do this by:

subtask 1:
method suggestWord(currentSubWord, file):
        1.  Read file
        2. Add to some sort of set (list without duplicates)
        3. compare currentSubWord with the set and check if a String in the set starts with currentSubWord
        4. return a new list containing possible suggested words

Not sure how this would look visually

**Comment by ThatsJustCheesy** [comment]
It's partially implemented [here](https://github.com/BushelScript/BushelScript/blob/aabe7b5977f6de36b9a6eb3768085acf46b3ab0e/BushelScript%20Editor/BushelScript%20Editor/Documents/DocumentVC.swift#L528), but I'm not entirely happy with it. Try hitting Escape with that checkbox enabled in [this build](https://github.com/BushelScript/BushelScript/suites/3098598980/artifacts/70814176). The current behaviour is to suggest matching words that appear before the current word, in reverse order, followed by matching words that appear after the current word. IMO, ideally, it would suggest words in more or less the same order as [TextMate 2 does](https://macromates.com/blog/2012/clever-completion/).

It's implemented essentially as you specified. Fair warning, Strings in Swift can be pretty bizarre to work with. But feel free to try it :D I can also suggest tasks in the compiler if you prefer.

**Reply by apinanyogaratnam** [reply]
holy cow LOL that's sickk. the code looks scary tho, its probably better than how I planned to do it. This is my progress but ill just scrap it. 
```import Foundation

func readFile(fileURL) -> String {
    do {
        readString = try String(contentsOf: FileURL);
    } catch let error as NSError {
        print("Failed to read file.");
        print(error);
    }
    return greeting
}

func suggestWord(String) -> [String] {
    var setOfPossibleWords: Set = [];

    var readString = "";
}

**Reply by apinanyogaratnam** [reply]
Sure I'd love to do a task in the compiler but if possible make it an easy task lol
